### ISSUE TITLE
Overfit or wrong loss

### ISSUE THREAD

**Issue by ihmc3jn09hk** [pvnieo/SURFMNet-pytorch#2]
Hi @pvnieo , I am training the network with the `Faust_original` dataset while there is something weir happening. Within 50 `mini-batch`(term used in the paper) iterations, the loss goes up quickly. I tried `batch-sizes` 1,2,4,8 but none worked, e.g.
```console
python3 train.py -bs 8 --n-epochs 2 -d data/Faust_original/MAT
```
I have tried lowering the `lr` by order of 2, e.g. 1e-5, which does prolong the loss upto 6th epoches but the loss never went down below 600. (The paper mentioned a scale of 10 from there results). I am not sure it is a sign of overfit or something else from the implementation. Any suggestion?

My setup is:
Ubuntu 20.04
Python 3.8.5 
and
```console
Package           Version
----------------- -----------
numpy             1.19.4
Pillow            8.0.1
pip               20.2.4
scipy             1.5.4
setuptools        50.3.2
torch             1.7.0+cu110
torchaudio        0.7.0
torchvision       0.8.1+cu110
...
```

```console
#epoch:1, #batch:1, #iteration:1, loss:618366.5
#epoch:1, #batch:2, #iteration:2, loss:191922.375
#epoch:1, #batch:3, #iteration:3, loss:390996.5625
#epoch:1, #batch:4, #iteration:4, loss:155653.78125
#epoch:1, #batch:5, #iteration:5, loss:122857.890625
#epoch:1, #batch:6, #iteration:6, loss:56080.80859375
#epoch:1, #batch:7, #iteration:7, loss:54986.0859375
#epoch:1, #batch:8, #iteration:8, loss:38508.7421875
#epoch:1, #batch:9, #iteration:9, loss:34346.59375
#epoch:1, #batch:10, #iteration:10, loss:48555.078125
#epoch:1, #batch:11, #iteration:11, loss:30304.625
#epoch:1, #batch:12, #iteration:12, loss:29533.83984375
#epoch:1, #batch:13, #iteration:13, loss:31013.697265625
#epoch:1, #batch:14, #iteration:14, loss:32305.798828125
#epoch:1, #batch:15, #iteration:15, loss:26479.25
#epoch:1, #batch:16, #iteration:16, loss:21710.76171875
#epoch:1, #batch:17, #iteration:17, loss:20447.109375
#epoch:1, #batch:18, #iteration:18, loss:19567.962890625
#epoch:1, #batch:19, #iteration:19, loss:23133.2578125
#epoch:1, #batch:20, #iteration:20, loss:15480.0986328125
#epoch:1, #batch:21, #iteration:21, loss:16198.91796875
#epoch:1, #batch:22, #iteration:22, loss:17660.18359375
#epoch:1, #batch:23, #iteration:23, loss:14030.8515625
#epoch:1, #batch:24, #iteration:24, loss:8511.982421875   <---------------------------
#epoch:1, #batch:25, #iteration:25, loss:43280.13671875
#epoch:1, #batch:26, #iteration:26, loss:34185.2734375
#epoch:1, #batch:27, #iteration:27, loss:15525.109375
#epoch:1, #batch:28, #iteration:28, loss:11670.400390625
#epoch:1, #batch:29, #iteration:29, loss:18548.89453125
#epoch:1, #batch:30, #iteration:30, loss:13342.068359375
#epoch:1, #batch:31, #iteration:31, loss:24310.1484375
#epoch:1, #batch:32, #iteration:32, loss:18103.34765625
#epoch:1, #batch:33, #iteration:33, loss:12983.232421875
#epoch:1, #batch:34, #iteration:34, loss:15744.41015625
#epoch:1, #batch:35, #iteration:35, loss:19102.33203125
#epoch:1, #batch:36, #iteration:36, loss:14325.658203125
#epoch:1, #batch:37, #iteration:37, loss:15989.193359375
#epoch:1, #batch:38, #iteration:38, loss:22554.8984375
#epoch:1, #batch:39, #iteration:39, loss:26798.611328125
#epoch:1, #batch:40, #iteration:40, loss:42119.96875
#epoch:1, #batch:41, #iteration:41, loss:93560.6328125
#epoch:1, #batch:42, #iteration:42, loss:117597.265625
#epoch:1, #batch:43, #iteration:43, loss:254039.015625
#epoch:1, #batch:44, #iteration:44, loss:134116.859375
#epoch:1, #batch:45, #iteration:45, loss:158679.6875
#epoch:1, #batch:46, #iteration:46, loss:165269.21875
#epoch:1, #batch:47, #iteration:47, loss:162998.328125
#epoch:1, #batch:48, #iteration:48, loss:245227.140625
#epoch:1, #batch:49, #iteration:49, loss:232698.59375
#epoch:1, #batch:50, #iteration:50, loss:629152.125
#epoch:1, #batch:51, #iteration:51, loss:379238.6875
#epoch:1, #batch:52, #iteration:52, loss:311628.1875
#epoch:1, #batch:53, #iteration:53, loss:421314.84375
...
```

**Comment by pvnieo** [comment]
Hi @ihmc3jn09hk 

Thanks for your interest, I will update the surfmnet and Fmnet repo by the end of the week, to fix the existing bugs!

I will let you know!

**Comment by ihmc3jn09hk** [comment]
> Hi @ihmc3jn09hk
> 
> Thanks for your interest, I will update the surfmnet and Fmnet repo by the end of the week, to fix the existing bugs!
> 
> I will let you know!

Thanks alot.

**Comment by pvnieo** [comment]
Hi @ihmc3jn09hk 

Sorry for the late reply, I was quite busy these days.
The two repos (FMNET & SURFMNET) are up to date.

Please let me know if you have any problem!

**Comment by ihmc3jn09hk** [comment]
> Hi @ihmc3jn09hk
> 
> Sorry for the late reply, I was quite busy these days.
> The two repos (FMNET & SURFMNET) are up to date.
> 
> Please let me know if you have any problem!

@pvnieo Its much appreciated for the effort. Will test it out and see if the problem is gone. Before that, may I ask for some explanation of the changes.
I found the weights are changed quite significantly. What measurements have you made to get such results ?

```python
    parser.add_argument("--wb", type=float, default=1, help="Bijectivity penalty weight")
    parser.add_argument("--wo", type=float, default=1, help="Orthogonality penalty weight")
    parser.add_argument("--wl", type=float, default=1e-4, help="Laplacian commutativity penalty weight")
    parser.add_argument("--wd", type=float, default=1e3, help="Descriptor preservation via commutativity penalty weight")
```

I see the loss criterion is changed too. Any difference on the `actual` calculation or just implementation difference ?

```python
        criterion = FrobeniusLoss()  #Before  criterion = nn.MSELoss(reduction="mean")
```

**Comment by pvnieo** [comment]
well, this just changed the scale of the loss, I worked fine for me. But you could try with the values suggested in the paper.
Also, the new criterion is the one described in the paper, the mean is computed over the batch, meanwhile, before it was computed on all the functional map!

**Comment by pvnieo** [comment]
I'm closing the issue, if you still have other question, feel free to open it!
